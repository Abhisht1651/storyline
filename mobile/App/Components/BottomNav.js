// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { View, TouchableWithoutFeedback, Image } from 'react-native'
import styles from './Styles/BottomNavStyles'
import { Images } from '../Themes'

type Props = {
  newsfeed: boolean,
  search: boolean,
  activity: boolean,
  chats: boolean,
  profile: boolean,
  newsfeedonPress: PropTypes.func,
  searchonPress: PropTypes.func,
  activityonPress: PropTypes.func,
  chatsonPress: PropTypes.func,
  profileonPress: PropTypes.func
}

export const BottomNav = (props: Props) => {
  const {
    newsfeed,
    search,
    activity,
    chats,
    profile,
    newsfeedonPress,
    searchonPress,
    activityonPress,
    chatsonPress,
    profileonPress
  } = props
  return (
    <View style={styles.BottomNav}>
      <View style={styles.box}>
        <View style={styles.container}>
          <View style={styles.margin} />
          <View style={styles.icons}>
            <TouchableWithoutFeedback onPress={newsfeedonPress}>
              <View style={styles.iconspacing}>
                {newsfeed ? (
                  <Image source={Images.newsfeed_sel} style={styles.newsfeed} />
                ) : (
                  <Image source={Images.newsfeed} style={styles.newsfeed} />
                )}
              </View>
            </TouchableWithoutFeedback>
            <TouchableWithoutFeedback onPress={searchonPress}>
              <View style={styles.iconspacing}>
                {search ? (
                  <Image source={Images.search_sel} style={styles.searchsel} />
                ) : (
                  <Image source={Images.search} style={styles.search} />
                )}
              </View>
            </TouchableWithoutFeedback>
            <TouchableWithoutFeedback onPress={activityonPress}>
              <View style={styles.iconspacing}>
                {activity ? (
                  <Image source={Images.friendsactivity_sel} style={styles.friendsactivity} />
                ) : (
                  <Image source={Images.friendsactivity} style={styles.friendsactivity} />
                )}
              </View>
            </TouchableWithoutFeedback>
            <TouchableWithoutFeedback onPress={chatsonPress}>
              <View style={styles.iconspacing}>
                {chats ? (
                  <Image source={Images.chat_sel} style={styles.chat} />
                ) : (
                  <Image source={Images.chat} style={styles.chat} />
                )}
              </View>
            </TouchableWithoutFeedback>
            <TouchableWithoutFeedback onPress={profileonPress}>
              <View style={styles.iconspacing}>
                {profile ? (
                  <Image source={Images.profile_sel} style={styles.profile} />
                ) : (
                  <Image source={Images.profile} style={styles.profile} />
                )}
              </View>
            </TouchableWithoutFeedback>
          </View>
          <View style={styles.margin} />
        </View>
      </View>
    </View>
  )
}

BottomNav.defaultProps = {
  newsfeed: false,
  search: false,
  activity: false,
  chats: false,
  profile: false,
  newsfeedonPress: () => {},
  searchonPress: () => {},
  activityonPress: () => {},
  chatsonPress: () => {},
  profileonPress: () => {}
}
