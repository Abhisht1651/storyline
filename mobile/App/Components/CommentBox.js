// @flow

import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import styles from './Styles/CommentBoxStyle'
import images from '../Themes/Images'
import DropDown from './Dropdowns/DropDown3'

type Props = {
  name: string,
  commentText: string,
  replyCount: number,
  isReply: boolean,
  boostCount: number,
  selectionHandler: (arg: number) => void,
  onClickBoost: PropTypes.func,
  onClickReply: PropTypes.func,
  time: string
}

const commentBox = (props: Props) => {
  const {
    name,
    commentText,
    replyCount,
    isReply,
    boostCount,
    selectionHandler,
    onClickBoost,
    onClickReply,
    time
  } = props
  const { replyIcon, boostIcon } = images
  return (
    <View style={styles.container}>
      <View style={styles.commentBoxContainer}>
        <View style={styles.commentBoxHeaderContainer}>
          <Text style={styles.nameText}>{name}</Text>
          <View style={styles.headerRight}>
            <Text style={styles.daysText}>{time}</Text>
            <View style={styles.dropDown}>
              <DropDown
                getOption={selectionHandler}
                headerTitle="Options"
                content={['Edit Comment', 'Delete Comment']}
              />
            </View>
          </View>
        </View>
        <View style={styles.commentTextContainer}>
          <Text style={styles.commentText}>{commentText}</Text>
        </View>
        <View style={styles.actionContainer}>
          {!isReply ? (
            <TouchableOpacity onPress={onClickReply} style={styles.replyContainer}>
              <Image source={replyIcon} style={styles.replyIcon} />
              <Text style={styles.actionText}> {replyCount !== 0 ? replyCount : null} Reply</Text>
            </TouchableOpacity>
          ) : null}
          <TouchableOpacity onPress={onClickBoost} style={styles.boostContainer}>
            <Image source={boostIcon} />
            <Text style={styles.actionText}> {boostCount} Boosts</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

commentBox.defaultProps = {
  name: 'Jammie Rice',
  commentText:
    'When Robert Mueller broke his silence in May, his main point was that his long-awaited report spoke for itself.',
  replyCount: 1,
  isReply: false,
  boostCount: 450,
  selectionHandler: () => {},
  onClickBoost: () => {},
  onClickReply: () => {},
  time: '32 Days'
}

export default commentBox
