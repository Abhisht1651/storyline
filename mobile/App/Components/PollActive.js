// @flow
import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { View, Text, Image, TouchableOpacity } from 'react-native'

// Styles
import styles from './Styles/PollActiveStyle'
import images from '../Themes/Images'

// Components
import { MediumThinnerButton } from './Buttons/MediumThinnerButton'

type Props = {
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>,
  option: Array<string>,
  comments: number,
  onPressVote: PropTypes.func,
  onPressShare: PropTypes.func,
  onPressReact: PropTypes.func,
  onPressRaven: PropTypes.func,
  onPressComment: PropTypes.func,
  onPressComments: PropTypes.func
}

const pollActive = (props: Props) => {
  const { pollActiveIcon, pollInactiveIcon, commentIcon, shareIcon, ravenIcon, reactIcon } = images
  const {
    question,
    votes,
    timeLeft,
    isPollActive,
    voterIcons,
    option,
    comments,
    onPressVote,
    onPressReact,
    onPressRaven,
    onPressShare,
    onPressComment,
    onPressComments
  } = props
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'

  const [optionSelected, setOptionSelected] = useState(0)

  const optionSelectFunction = (currentOption) => {
    if (currentOption === optionSelected) {
      setOptionSelected(0)
      return
    }
    setOptionSelected(currentOption)
  }

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={styles.container}>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterAndVotesContainer}>
        <View style={styles.voterContainer}>
          <View style={styles.voterIconContainer}>{voterIconsElement}</View>
          <View style={styles.votesContainer}>
            <Text style={styles.votes}>{votes}K votes</Text>
          </View>
        </View>
        <View style={styles.timeContainer}>
          <View style={styles.timeIconContainer}>
            <Image source={pollIcon} style={styles.pollIcon} />
          </View>
          <View style={styles.timeTextContainer}>
            <Text style={styles.timeText}>{pollText}</Text>
          </View>
        </View>
      </View>
      <View style={styles.optionsContainer}>
        <TouchableOpacity style={styles.optionContainer} onPress={() => optionSelectFunction(1)}>
          <View style={styles.option}>
            {optionSelected === 1 ? <View style={styles.optionFill} /> : null}
          </View>
          <View style={styles.optionTextContainer}>
            <Text style={styles.optionText}>{option[0]}</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity style={styles.optionContainer} onPress={() => optionSelectFunction(2)}>
          <View style={styles.option}>
            {optionSelected === 2 ? <View style={styles.optionFill} /> : null}
          </View>
          <View style={styles.optionTextContainer}>
            <Text style={styles.optionText}>{option[1]}</Text>
          </View>
        </TouchableOpacity>
      </View>
      <View style={styles.optionsContainer}>
        <TouchableOpacity style={styles.optionContainer} onPress={() => optionSelectFunction(3)}>
          <View style={styles.option}>
            {optionSelected === 3 ? <View style={styles.optionFill} /> : null}
          </View>
          <View style={styles.optionTextContainer}>
            <Text style={styles.optionText}>{option[2]}</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity style={styles.optionContainer} onPress={() => optionSelectFunction(4)}>
          <View style={styles.option}>
            {optionSelected === 4 ? <View style={styles.optionFill} /> : null}
          </View>
          <View style={styles.optionTextContainer}>
            <Text style={styles.optionText}>{option[3]}</Text>
          </View>
        </TouchableOpacity>
      </View>
      <View style={styles.buttonContainer}>
        <MediumThinnerButton title="Vote" outline onPress={onPressVote} />
      </View>
      <View style={styles.commentContainer}>
        <TouchableOpacity onPress={onPressComments}>
          <Text style={styles.commentText}>{comments}K Comments</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressComment} style={styles.commentButtonContainer}>
          <Image source={commentIcon} style={styles.commentIcon} />
          <Text style={styles.commentButtonText}>Comment</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.actionIconsContainer}>
        <TouchableOpacity onPress={onPressShare}>
          <Image source={shareIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressRaven}>
          <Image source={ravenIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressReact}>
          <Image source={reactIcon} style={styles.actionIcon} />
        </TouchableOpacity>
      </View>
    </View>
  )
}

pollActive.defaultProps = {
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3],
  option: ['India', 'West Indies', 'Australia', 'England'],
  comments: 154,
  onPressVote: () => {},
  onPressReact: () => {},
  onPressRaven: () => {},
  onPressShare: () => {},
  onPressComment: () => {},
  onPressComments: () => {}
}

export default pollActive
