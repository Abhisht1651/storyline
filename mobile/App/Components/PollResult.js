// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, Image, TouchableOpacity } from 'react-native'

// Styles
import styles from './Styles/PollResultStyle'
import images from '../Themes/Images'

type Props = {
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>,
  progressBar: Array<{ name: string, percentage: string }>,
  participated: boolean,
  comments: number,
  activeProgressBar: number,
  onPressShare: PropTypes.func,
  onPressReact: PropTypes.func,
  onPressRaven: PropTypes.func,
  onPressComment: PropTypes.func,
  onPressComments: PropTypes.func
}

const pollResult = (props: Props) => {
  const {
    question,
    votes,
    timeLeft,
    isPollActive,
    voterIcons,
    progressBar,
    participated,
    comments,
    activeProgressBar,
    onPressReact,
    onPressRaven,
    onPressShare,
    onPressComment,
    onPressComments
  } = props
  const { pollActiveIcon, pollInactiveIcon, commentIcon, shareIcon, ravenIcon, reactIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'
  const fillColor = participated ? '#7B46E4' : '#47C3F4'

  const progressBarStyles = [
    { ...styles.progressFill, width: progressBar[0].percentage },
    { ...styles.progressFill, width: progressBar[1].percentage },
    { ...styles.progressFill, width: progressBar[2].percentage },
    { ...styles.progressFill, width: progressBar[3].percentage }
  ]

  progressBarStyles[activeProgressBar] = {
    ...progressBarStyles[activeProgressBar],
    backgroundColor: fillColor
  }

  const progressBarActivePercentageText = { ...styles.progressBarPercentageText, color: '#FFFFFF' }
  const progressBarActiveText = { ...styles.progressBarText, color: '#FFFFFF' }

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={styles.container}>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterAndVotesContainer}>
        <View style={styles.voterContainer}>
          <View style={styles.voterIconContainer}>{voterIconsElement}</View>
          <View style={styles.votesContainer}>
            <Text style={styles.votes}>{votes}K votes</Text>
          </View>
        </View>
        <View style={styles.timeContainer}>
          <View style={styles.timeIconContainer}>
            <Image source={pollIcon} style={styles.pollIcon} />
          </View>
          <View style={styles.timeTextContainer}>
            <Text style={styles.timeText}>{pollText}</Text>
          </View>
        </View>
      </View>

      <View style={styles.progressBarContainer}>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[0]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 0
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[0].percentage}
            </Text>
            <Text style={activeProgressBar === 0 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[0].name}
            </Text>
          </View>
          {activeProgressBar === 0 && participated ? (
            <View style={styles.progressBarVoterIconContainer}>
              <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
            </View>
          ) : null}
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[1]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 1
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[1].percentage}
            </Text>
            <Text style={activeProgressBar === 1 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[1].name}
            </Text>
            {activeProgressBar === 1 && participated ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[2]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 2
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[2].percentage}
            </Text>
            <Text style={activeProgressBar === 2 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[2].name}
            </Text>
            {activeProgressBar === 2 && participated ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[3]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 3
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[3].percentage}
            </Text>
            <Text style={activeProgressBar === 3 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[3].name}
            </Text>
            {activeProgressBar === 3 && participated ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
      </View>

      <View style={styles.commentContainer}>
        <TouchableOpacity onPress={onPressComments}>
          <Text style={styles.commentText}>{comments}K Comments</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressComment} style={styles.commentButtonContainer}>
          <Image source={commentIcon} style={styles.commentIcon} />
          <Text style={styles.commentButtonText}>Comment</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.actionIconsContainer}>
        <TouchableOpacity onPress={onPressShare}>
          <Image source={shareIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressRaven}>
          <Image source={ravenIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressReact}>
          <Image source={reactIcon} style={styles.actionIcon} />
        </TouchableOpacity>
      </View>
    </View>
  )
}

pollResult.defaultProps = {
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3],
  progressBar: [
    { name: 'India', percentage: '53%' },
    { name: 'Australia', percentage: '12%' },
    { name: 'West Indies', percentage: '6%' },
    { name: 'England', percentage: '29%' }
  ],
  participated: true,
  comments: 154,
  activeProgressBar: 0,
  onPressReact: () => {},
  onPressRaven: () => {},
  onPressShare: () => {},
  onPressComment: () => {},
  onPressComments: () => {}
}

export default pollResult
