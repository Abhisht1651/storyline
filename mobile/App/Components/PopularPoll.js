// @flow
import React from 'react'
import { View, Text, Image } from 'react-native'

// Styles
import styles from './Styles/PopularPollStyle'
import images from '../Themes/Images'

type Props = {
  border: boolean,
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>,
  progressBar: Array<{ name: string, percentage: string }>,
  activeProgressBar: number
}

const popularPoll = (props: Props) => {
  const {
    border,
    question,
    votes,
    timeLeft,
    isPollActive,
    voterIcons,
    progressBar,
    activeProgressBar
  } = props
  const { pollActiveIcon, pollInactiveIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'
  const fillColor = isPollActive ? '#7B46E4' : '#47C3F4'

  const progressBarStyles = [
    { ...styles.progressFill, width: progressBar[0].percentage },
    { ...styles.progressFill, width: progressBar[1].percentage },
    { ...styles.progressFill, width: progressBar[2].percentage },
    { ...styles.progressFill, width: progressBar[3].percentage }
  ]

  progressBarStyles[activeProgressBar] = {
    ...progressBarStyles[activeProgressBar],
    backgroundColor: fillColor
  }

  const progressBarActivePercentageText = { ...styles.progressBarPercentageText, color: '#FFFFFF' }
  const progressBarActiveText = { ...styles.progressBarText, color: '#FFFFFF' }

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={border ? styles.container : styles.containerWithoutBorder}>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterAndVotesContainer}>
        <View style={styles.voterContainer}>
          <View style={styles.voterIconContainer}>{voterIconsElement}</View>
          <View style={styles.votesContainer}>
            <Text style={styles.votes}>{votes}K votes</Text>
          </View>
        </View>
        <View style={styles.timeContainer}>
          <View style={styles.timeIconContainer}>
            <Image source={pollIcon} style={styles.pollIcon} />
          </View>
          <View style={styles.timeTextContainer}>
            <Text style={styles.timeText}>{pollText}</Text>
          </View>
        </View>
      </View>

      <View style={styles.progressBarContainer}>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[0]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 0
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[0].percentage}
            </Text>
            <Text style={activeProgressBar === 0 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[0].name}
            </Text>
          </View>
          {activeProgressBar === 0 ? (
            <View style={styles.progressBarVoterIconContainer}>
              <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
            </View>
          ) : null}
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[1]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 1
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[1].percentage}
            </Text>
            <Text style={activeProgressBar === 1 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[1].name}
            </Text>
            {activeProgressBar === 1 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[2]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 2
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[2].percentage}
            </Text>
            <Text style={activeProgressBar === 2 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[2].name}
            </Text>
            {activeProgressBar === 2 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[3]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 3
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[3].percentage}
            </Text>
            <Text style={activeProgressBar === 3 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[3].name}
            </Text>
            {activeProgressBar === 3 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
      </View>
    </View>
  )
}

popularPoll.defaultProps = {
  border: true,
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3],
  progressBar: [
    { name: 'India', percentage: '53%' },
    { name: 'Australia', percentage: '12%' },
    { name: 'West Indies', percentage: '6%' },
    { name: 'England', percentage: '29%' }
  ],
  activeProgressBar: 0
}

export default popularPoll
