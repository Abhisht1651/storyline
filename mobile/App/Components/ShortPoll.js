// @flow
import React from 'react'
import { View, Text, Image } from 'react-native'

// Style
import styles from './Styles/ShortPollStyle'
import images from '../Themes/Images'

type Props = {
  border: boolean,
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>
}

const shortPoll = (props: Props) => {
  const { border, question, votes, timeLeft, isPollActive, voterIcons } = props
  const { pollActiveIcon, pollInactiveIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={border ? styles.container : styles.containerWithoutBorder}>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterAndVotesContainer}>
        <View style={styles.voterContainer}>
          <View style={styles.voterIconContainer}>{voterIconsElement}</View>
          <View style={styles.votesContainer}>
            <Text style={styles.votes}>{votes}K votes</Text>
          </View>
        </View>
        <View style={styles.timeContainer}>
          <View style={styles.timeIconContainer}>
            <Image source={pollIcon} style={styles.pollIcon} />
          </View>
          <View style={styles.timeTextContainer}>
            <Text style={styles.timeText}>{pollText}</Text>
          </View>
        </View>
      </View>
    </View>
  )
}

shortPoll.defaultProps = {
  border: true,
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
}

export default shortPoll
