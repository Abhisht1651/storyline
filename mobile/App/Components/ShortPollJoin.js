// @flow

import React from 'react'
import { View, Text, Image } from 'react-native'

// Styles
import styles from './Styles/ShortPollJoinStyle'
import images from '../Themes/Images'

type Props = {
  border: boolean,
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>
}

const shortPollJoin = (props: Props) => {
  const { border, question, votes, timeLeft, isPollActive, voterIcons } = props
  const { pollActiveIcon, pollInactiveIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={border ? styles.container : styles.containerWithoutBorder}>
      <View style={styles.joinTextContainer}>
        <Text style={styles.joinText}>Join to this poll!</Text>
      </View>
      <View style={styles.divider} />
      <View style={styles.timeContainer}>
        <View style={styles.timeIconContainer}>
          <Image source={pollIcon} style={styles.pollIcon} />
        </View>
        <View style={styles.timeTextContainer}>
          <Text style={styles.timeText}>{pollText}</Text>
        </View>
      </View>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterContainer}>
        <View style={styles.voterIconContainer}>{voterIconsElement}</View>
        <View style={styles.votesContainer}>
          <Text style={styles.votes}>{votes}K have voted</Text>
        </View>
      </View>
    </View>
  )
}

shortPollJoin.defaultProps = {
  border: true,
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3]
}

export default shortPollJoin
