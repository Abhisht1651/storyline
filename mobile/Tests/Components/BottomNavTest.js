/* eslint-disable prettier/prettier */
import React from 'react'
import { shallow } from 'enzyme'
import { Image, TouchableWithoutFeedback } from 'react-native'
import { BottomNav } from '../../App/Components/BottomNav'

// Test for BottomNav (snapshot and props)

describe('BottomNav', () => {
  it('BottomNav component renders correctly', () => {
    const tree = shallow(<BottomNav />)
    expect(tree).toMatchSnapshot()
  })
  it('render 5 TouchableWithoutFeedback', () => {
    const wrapper = shallow(<BottomNav />)
    expect(wrapper.find(TouchableWithoutFeedback)).toHaveLength(5)
  })
  it('render 5 Image', () => {
    const wrapper = shallow(<BottomNav />)
    expect(wrapper.find(Image)).toHaveLength(5)
  })
})
