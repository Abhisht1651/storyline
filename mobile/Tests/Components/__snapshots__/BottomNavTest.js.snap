// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNav BottomNav component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#E6ECED",
        "borderTopWidth": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 375,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 25,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 325,
            "flexDirection": "row",
          }
        }
      >
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": 52,
                  "width": 64,
                }
              }
            />
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": 44,
                  "width": 52,
                }
              }
            />
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": 48,
                  "width": 66,
                }
              }
            />
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": 48,
                  "width": 53.99999999999999,
                }
              }
            />
          </View>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": 52,
                  "width": 50,
                }
              }
            />
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={
          Object {
            "flex": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
